Neos.Neos/Classes/Command/WorkspaceCommandController.php

```diff
@@ -260,19 +260,21 @@ public function deleteCommand(string $workspace, bool $force = false, string $co
                 $this->outputLine(
                     'Did not delete workspace "%s" because it contains %s unpublished node(s).'
                         . ' Use --force to delete it nevertheless.',
-                    [$workspaceName, $nodesCount]
+                    [$workspaceName->value, $nodesCount]
                 );
                 $this->quit(5);
             }
-            $this->discardCommand($workspaceName);
+            $contentRepository->handle(
+                DiscardWorkspace::create($workspaceName)
+            )->block();
         }
 
-        $this->workspaceCommandHandler->handlePublishIndividualNodesFromWorkspace()
-
-
-        $this->workspaceFinder->remove($workspace);
-        $this->outputLine('Deleted workspace "%s"', [$workspaceName]);
-        */
+        $contentRepository->handle(
+            DeleteWorkspace::create(
+                $workspaceName
+            )
+        )->block();
```

Comment:
The block is not necessary here, isn't it?

Reply:
well dont we want to make sure that its actually deleted by the time we exit 0 ?

Reply:
otherwise it would be trigger deletion, but what if this fails and how log this? ^^

Reply:
It will be deleted eventually.

After firing the command, nothing shoud(tm) fail anymore.

Reply:
In this case it doesn't matter if it is async or sync. But I think we need to learn to choose the block wisely, otherwise we could make everything sync again as before. 🤷‍♂️

Reply:
yes your completely right. that makes actually perfect sense. Lets ask in the weekly ^^

Reply:
In the weekly we concluded that we want to block to align to the behaviour of the other cli  commands